File: TUnit.Templates/content/TUnit.VB/Data/DataGenerator.vb
Imports System
Imports System.Collections.Generic
Imports TUnit.Core

Namespace Data

    Public Class DataGenerator
        Inherits DataSourceGeneratorAttribute(Of Integer, Integer, Integer)

        Public Overrides Iterator Function GenerateDataSources(dataGeneratorMetadata As DataGeneratorMetadata) As IEnumerable(Of Func(Of (Integer, Integer, Integer)))
            Yield Function() (1, 1, 2)
            Yield Function() (1, 2, 3)
            Yield Function() (4, 5, 9)
        End Function

    End Class

End Namespace

File: TUnit.Templates/content/TUnit.FSharp/Data/DataGenerator.fs
namespace TestProject.Data

open System
open System.Collections.Generic
open TUnit.Core

type DataGenerator() =
    inherit DataSourceGeneratorAttribute<int, int, int>()

    override _.GenerateDataSources(_: DataGeneratorMetadata) : IEnumerable<Func<struct (int * int * int)>> =
        seq {
            yield Func<struct (int * int * int)>(fun () -> struct (1, 1, 2))
            yield Func<struct (int * int * int)>(fun () -> struct (1, 2, 3))
            yield Func<struct (int * int * int)>(fun () -> struct (4, 5, 9))
        }

